This hack hardcodes pxa27x-udc to ether_gadget. This is just a temp workaround.
Index: linux-2.6.21/drivers/usb/gadget/ether.c
===================================================================
--- linux-2.6.21.orig/drivers/usb/gadget/ether.c	2007-09-23 09:32:51.000000000 -0300
+++ linux-2.6.21/drivers/usb/gadget/ether.c	2007-09-23 09:32:52.000000000 -0300
@@ -2697,10 +2697,16 @@
 MODULE_LICENSE ("GPL");
 
 
-static int __init init (void)
+// Alex add FIXME
+int usb_ether_init (void)
 {
 	return usb_gadget_register_driver (&eth_driver);
 }
+
+static int __init init (void)
+{
+//	return usb_gadget_register_driver (&eth_driver);
+}
 module_init (init);
 
 static void __exit cleanup (void)
Index: linux-2.6.21/drivers/usb/gadget/pxa27x_udc.c
===================================================================
--- linux-2.6.21.orig/drivers/usb/gadget/pxa27x_udc.c	2007-09-23 09:32:51.000000000 -0300
+++ linux-2.6.21/drivers/usb/gadget/pxa27x_udc.c	2007-09-23 09:32:52.000000000 -0300
@@ -2227,6 +2227,8 @@
 	udc_init_ep(udc);
 	udc_reinit(udc);
 
+	usb_ether_init();  // Alex add FIXME
+
 	/* irq setup after old hardware state is cleaned up */
 	retval = request_irq(IRQ_USB, pxa27x_udc_irq,
 			SA_INTERRUPT, driver_name, udc);
